partcad/src/partcad/logging_ansi_terminal.py

```
@@ -17,6 +18,32 @@
 
 from .logging import ops, error
 
+
+class TimeSortedActions:
+    def __init__(self) -> None:
+        self.actions: dict = {}  # Use a regular dictionary to store data
+        self.sorted_action_names: list[str] = []
+
+    def __setitem__(self, key: str, value) -> None:
```

Ruff (0.8.2)

Missing type annotation for function argument `value`

(ANN001)